pkg/clients/aws/route53.go

```diff
@@ -611,16 +611,29 @@ func getSTSCredentials(reqLogger logr.Logger, client *sts.STS, roleArn string, e
 		if err == nil {
 			break
 		}
-		if i == (assumeRolePollingRetries - 1) {
-			return nil, fmt.Errorf("timed out while assuming role %s", roleArn)
+		// Error was returned
+		if i >= (assumeRolePollingRetries - 1) {
+			// If we're out of retries, return the wrapped error
+			return nil, fmt.Errorf("failed to assumeRole, no retries left: %w", err)
 		}
-	}
-	if err != nil {
 		// Log AWS error
 		if aerr, ok := err.(awserr.Error); ok {
 			reqLogger.Error(aerr, "New AWS Error while getting STS credentials,\nAWS Error Code: %s,\nAWS Error Message: %s", aerr.Code(), aerr.Message())
+			// Handle specific AWS errors
+			switch aerr.Code() {
+			case sts.ErrCodeExpiredTokenException:
+				// For expired token errors, stop retrying as it indicates an issue with credentials
+				return nil, fmt.Errorf("expired token error: %w", aerr)
+
+			default:
+				// Handle other AWS errors as needed
+				reqLogger.Info("\"%w\", Unhandled AWS error: %s", aerr, aerr.Code())
+				// return nil, fmt.Errorf("AWS error: %w", aerr)
+				continue
+			}
 		}
-		return &sts.AssumeRoleOutput{}, err
+		// If we still have retries, log the failure and try again
+		reqLogger.Error(err, "failed to assumeRole (attempt %d out of %d): %v", i, assumeRolePollingRetries, err)
```

Comment:
Two options here:

1) We leave this as an Error log type but we can remove the `: %v` part of the log message since it's already going to spit out the `err` as part of the Error logger
2) We change this to be an `reqLogger.Info()` function since it's just informational that we're retrying, still printing the error to the log message for any potential debugging, but not printing out the full stack trace that `reqLogger.Error` would give us.